routing/missioncontrol.go

```diff
@@ -299,6 +305,36 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 	return bandwidthHints, nil
 }
 
+// updateEdgePolicy updates the channel edge policy parameters if both channel
+// ID and direction match those of the edge locator.
+func updateEdgePolicy(el *edgeLocator, up *lnwire.ChannelUpdate,
+	ep *channeldb.ChannelEdgePolicy) {
+
+	if ep.ChannelID == el.channelID &&
+		uint8(ep.Flags&lnwire.ChanUpdateDirection) == el.direction {
+
+		ep.FeeBaseMSat = lnwire.MilliSatoshi(up.BaseFee)
+		ep.FeeProportionalMillionths = lnwire.MilliSatoshi(up.FeeRate)
+		ep.TimeLockDelta = up.TimeLockDelta
+		ep.MinHTLC = lnwire.MilliSatoshi(up.HtlcMinimumMsat)
+	}
+}
+
+// UpdateEdgePolicy updates edge policy of the routing hints kept in the
+// payment session.
+func (p *paymentSession) UpdateEdgePolicy(e *edgeLocator,
+	update *lnwire.ChannelUpdate) {
+
+	log.Debugf("Updating edge %v policy in Mission Control", e)
+
+	// Run over hints and update policy if edge id and direction match.
+	for _, edges := range p.additionalEdges {
+		for _, edge := range edges {
+			updateEdgePolicy(e, update, edge)
```

Comment:
This is a bit hard to follow IMO. A better approach could be 
```go
for _, edges := range p.additionalEdges {
	for _, edge := range edges {
                if edge.ChannelID != e.ChannelID {
                      continue
                }

                if ..!checkDirection... {
                      continue
                }
 
                 ... update policy ....
        }
}
```
Also I don't think you need to pass the `edgeLocator` into this method? Might have everything you need in the `ChannelUpdate`. If not just pass the `pubKey` as in `applyChannelUpdate`.

Reply:
Done